lib/go/thrift/http_client_test.go

```diff
@@ -103,6 +105,54 @@ func TestHttpCustomClientPackageScope(t *testing.T) {
 	}
 }
 
+func TestHTTPClientFlushesRequestBufferOnErrors(t *testing.T) {
+	var (
+		write1 = []byte("write 1")
+		write2 = []byte("write 2")
+	)
+
+	l, addr := HttpClientSetupForTest(t)
+	if l != nil {
+		defer l.Close()
+	}
+	trans, err := NewTHttpPostClient("http://" + addr.String())
+	if err != nil {
+		t.Fatalf("Unable to connect to %s: %s", addr.String(), err)
```

Comment:
This is nitpicking and also a problem already exists in the old test code, but using `%s` on errors is the same as using `%v` on errors except that `%s` handles it much more poorer when error is nil (https://play.golang.org/p/yHovIp3m37B). I'd rather just use `%v` with errors instead.